config/ogdl/src/main/java/org/apache/shiro/config/ogdl/ReflectionBuilder.java

```diff
@@ -290,7 +293,9 @@ private String parseBeanId(String lhs) {
         }
 
         //SHIRO-413: init method must be called for constructed objects that are Initializable
-        LifecycleUtils.init(objects.values());
+        //SHIRO-778: onInit method on AuthenticatingRealm is called twice
+        objects.keySet().stream().filter(key -> !Optional.ofNullable(kvPairs).orElseGet(HashMap::new).keySet().contains(key))
+                .forEach(key -> LifecycleUtils.init(objects.get(key)));
```

Comment:
Having worked with a lot of lambdas, two suggestions here:
* Line break after `.stream()` and  each operation (after `.filter()` and `.forEach()` in this case).
* refactor out the `.filter()` contents. It is much easier to read a method name here than to grasp the contents of the whole filter operation and where parentheses actually ends.

Reply:
ok for the break line. However for the refactor out the filter() content, I'm not fully agree, the content is short.

Reply:
Ok, I got an better idea anyway.
At the moment, you are creating a new map for each key of kvPairs is empty! That's a lot of overhead!

At least move this part to its own statement on the previous line:
`Optional.ofNullable(kvPairs).orElseGet(HashMap::new)`.

This will avoid creating 10 new hashmaps of you have 10 keys in objects. And it's much easier to read the lambda (if !map.contains)